tools/RAiDER/utilFcns.py

```diff
@@ -1135,3 +1136,59 @@ def transform_coords(proj1, proj2, x, y):
     """
     transformer = Transformer.from_crs(proj1, proj2, always_xy=True)
     return transformer.transform(x, y)
+
+
+def get_nearest_wmtimes(t0, time_delta):
+    """"
+    Get the nearest two available times to the requested time given a time step
+
+    Args:
+        t0         - user-requested Python datetime 
+        time_delta  - time interval of weather model
+    
+    Returns:
+        tuple: list of datetimes representing the one or two closest 
+        available times to the requested time
+    
+    Example: 
+    >>> import datetime
+    >>> from RAiDER.utilFcns import get_nearest_wmtimes
+    >>> t0 = datetime.datetime(2020,1,1,11,35,0)
+    >>> get_nearest_wmtimes(t0, 3)
+     (datetime.datetime(2020, 1, 1, 9, 0), datetime.datetime(2020, 1, 1, 12, 0))
+    """    
+    # get the closest time available
+    tclose = round_time(t0, roundTo = time_delta * 60 *60)
+
+    # Just calculate both options and take the closest
+    t2_1 = tclose + timedelta(hours=time_delta)
+    t2_2 = tclose - timedelta(hours=time_delta)
+    t2 = [t2_1 if get_dt(t2_1, t0) < get_dt(t2_2, t0) else t2_2][0]
+
+    # If you're within 5 minutes just take the closest time
+    if get_dt(tclose, t0) < _THRESHOLD_SECONDS:
+        return [tclose]
+    else:
+        if t2 > tclose:
+            return [tclose, t2]
+        else:
+            return [t2, tclose]
+
```

Comment:
such an assumption should be documented. Once might question by 5 min why not 30 min; why interpolate at all?

Reply:
@dbekaert yes exactly. I should update the CHANGELOG. The idea is is that if you are within some small increment of a date, there is no reason to interpolate. The increment is set by `_THRESHOLD_SECONDS`. E.g. if you're within 10 seconds, you save time and numerical issues by simply taking the closest time.